scripts/docs-deploy/deploy-ci-push.ts

```diff
@@ -0,0 +1,112 @@
+
+  if (branchName === active.next.branchName) {
+    const major = active.next.version.major;
+    const targets = [{projectId, description, site: sites.next}];
+
+    // If the next release train is for a new major that is not published as part of the
+    // other active release trains, we also publish to e.g. `v14.material.angular.io`.
+    if (major > (active.releaseCandidate ?? active.latest).version.major) {
+      targets.push({projectId, description, site: sites.forMajor(major)});
+    }
+
+    await buildAndDeployWithSnapshots(authToken, major, targets);
+    return;
+  }
+
+  if (branchName === active.latest.branchName) {
+    const major = active.latest.version.major;
+    const targets = [
+      {projectId, description, site: sites.stable},
+      {projectId, description, site: sites.forMajor(major)},
+    ];
+
+    // If there is no active RC train, we also push the current stable to the `rc` site.
+    // TODO: This can be improved by using redirects from rc -> stable.
+    if (active.releaseCandidate === null) {
+      targets.push({projectId, description, site: sites.rc});
+    }
+
+    await buildAndDeployWithSnapshots(authToken, major, targets, {
+      prebuild: docsRepoDir => updateVersionsFile(docsRepoDir, active),
+    });
+    return;
+  }
+
+  if (branchName === active.releaseCandidate?.branchName) {
+    const major = active.releaseCandidate.version.major;
+    const targets = [{projectId, description, site: sites.rc}];
+
+    // If the RC is for a new major that neither `latest`, nor `next` publish,
+    // we will also push the dedicated major like `v13.material.angular.io`.
+    if (major > active.latest.version.major && major < active.next.version.major) {
+      targets.push({projectId, description, site: sites.forMajor(major)});
+    }
+
+    await buildAndDeployWithSnapshots(authToken, major, targets);
+    return;
+  }
+
+  // In other cases, we are potentially deploying an archived major, regardless
+  // of LTS being active or not.
+  if (!isVersionBranch(branchName)) {
+    console.log('Skipping deployment as the current branch is not a version branch.');
+    return;
+  }
+
+  const branchVersion = getVersionForVersionBranch(branchName)!;
+  const branchesForMajor = await getBranchesForMajorVersions(repo, [branchVersion.major]);
+
+  // The `branchesForMajor` array will hold the most recent version branch for current major.
+  // If the latest branch does not match the current one, we know that this is not the
+  // most recent minor and should not be re-deployed to e.g. `<major>.material.angular.io`.
+  if (branchesForMajor[0].name !== branchName) {
+    console.log(
+      `Skipping deployment as version branch is not the most recent ` +
+        `one for v${branchVersion.major}. Expected: ${branchesForMajor[0].name}`,
+    );
+    return;
+  }
+
+  const {parsed} = await getPackageJsonOfProject(projectDir);
+  const version = semver.parse(parsed.version)!;
```

Comment:
Couldn't you just use `branchVersion.major` (and skip computing `version`)?

Reply:
great idea 👍